pkg/utils/gitops/controller.go

```diff
@@ -0,0 +1,143 @@
+package gitops
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	routev1 "github.com/openshift/api/route/v1"
+	kubeCl "github.com/redhat-appstudio/e2e-tests/pkg/apis/kubernetes"
+	managedgitopsv1alpha1 "github.com/redhat-appstudio/managed-gitops/backend/apis/managed-gitops/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type SuiteController struct {
+	*kubeCl.K8sClient
+}
+
+func NewSuiteController(kube *kubeCl.K8sClient) (*SuiteController, error) {
+	return &SuiteController{
+		kube,
+	}, nil
+}
+
+func (h *SuiteController) CreateGitOpsDeployment(name string, namespace string, repoUrl string, repoPath string, repoRevision string) (*managedgitopsv1alpha1.GitOpsDeployment, error) {
+	gitOpsDeployment := &managedgitopsv1alpha1.GitOpsDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: managedgitopsv1alpha1.GitOpsDeploymentSpec{
+			Source: managedgitopsv1alpha1.ApplicationSource{
+				RepoURL:        repoUrl,
+				Path:           repoPath,
+				TargetRevision: repoRevision,
+			},
+			Type: managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated,
+		},
+	}
+
+	err := h.KubeRest().Create(context.TODO(), gitOpsDeployment)
+	if err != nil {
+		return nil, err
+	}
+	return gitOpsDeployment, nil
+}
+
+// DeleteGitOpsDeployment deletes an gitops deployment from a given name and namespace
+func (h *SuiteController) DeleteGitOpsDeployment(name string, namespace string) error {
+	gitOpsDeployment := &managedgitopsv1alpha1.GitOpsDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+	return h.KubeRest().Delete(context.TODO(), gitOpsDeployment)
+}
+
+// GetComponentRoute returns the route for a given component name
+func (h *SuiteController) GetComponentRoute(componentName string, componentNamespace string) (*routev1.Route, error) {
```

Comment:
I think this function should be: GetOpenshiftRoute and should be in commonController.

GetComponentRoute(routeName string, routeNamespace string) (*routev1.Route, error) {